### ISSUE TITLE
10bit HEVC hardware acceleration broken after seeking

### ISSUE THREAD

**Issue by LeoNatan** [lhc70000/iina#1996]
**Edit:** This issue occurs only after seeking in IINA. Doesn't occur after seeking in mpv.

No hardware acceleration is used on a 2018 MBP.

**System and IINA version:**

- macOS 10.14
- IINA 1.0.0-beta4

**Expected behavior:**
Should use hardware acceleration

**Actual behavior:**
300% CPU usage

- Crash report:
```

```

- MPV log:
```
[   0.292][i][cplayer]  (+) Video --vid=1 (*) (hevc 3840x1608 23.976fps)
[   0.292][i][cplayer]  (+) Audio --aid=1 --alang=eng (*) 'Atmos 7.1' (truehd 8ch 48000Hz)
[   0.292][i][cplayer]      Audio --aid=2 --alang=eng 'AC-3 5.1' (ac3 6ch 48000Hz)
[   0.292][i][cplayer]      Subs  --sid=1 --slang=eng 'Stripped SRT' (subrip)
[   0.292][i][cplayer]      Subs  --sid=2 --slang=eng 'SDH SRT' (subrip)
[   0.292][i][cplayer]      Subs  --sid=3 --slang=eng 'SDH' (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=4 --slang=fre (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=5 --slang=spa (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=6 --slang=spa (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=7 --slang=por (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=8 --slang=ita (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=9 --slang=swe (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=10 --slang=nor (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=11 --slang=dan (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=12 --slang=fin (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=13 --slang=rus (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=14 --slang=ara (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=15 --slang=chi (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=16 --slang=chi (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=17 --slang=chi (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=18 --slang=kor (hdmv_pgs_subtitle)
[   0.292][i][cplayer]      Subs  --sid=19 --slang=tha (hdmv_pgs_subtitle)
[   0.293][v][vd] Container reported FPS: 23.976025
[   0.293][v][vd] Codec list:
[   0.293][v][vd]     hevc - HEVC (High Efficiency Video Coding)
[   0.293][v][vd] Opening decoder hevc
[   0.293][v][vd] Looking at hwdec hevc-videotoolbox...
[   0.293][v][vd] Trying hardware decoding via hevc-videotoolbox.
[   0.294][v][vd] Selected codec: hevc (HEVC (High Efficiency Video Coding))
[   0.294][v][vf] User filter list:
[   0.294][v][vf]   (empty)
[   0.294][v][ad] Codec list:
[   0.294][v][ad]     truehd - TrueHD
[   0.294][v][ad] Opening decoder truehd
[   0.294][v][ad] Detected 12 logical cores.
[   0.294][v][ad] Requesting 13 threads for decoding.
[   0.294][v][ad] Selected codec: truehd (TrueHD)
[   0.294][v][af] User filter list:
[   0.294][v][af]   (empty)
[   0.294][v][cplayer] Starting playback...
[   0.294][v][mkv] queuing seek to 5546.458000
[   0.294][v][cplayer] hr-seek, skipping to 5546.458000
[   0.294][v][mkv] execute seek (to 5546.458000 flags 32)
[   0.294][d][cache] request seek: 9266 <= to=15754296941 (cur=9266) <= 7349298  
[   0.294][v][cache] Dropping cache at pos 15754296941, cached range: 9266-7414834.
[   0.294][v][cache] Seeking underlying stream: 7414834 -> 15754296941
[   0.295][v][mkv] seek done
[   0.296][v][vd] Pixel formats supported by decoder: videotoolbox_vld yuv420p10le
[   0.296][v][vd] Codec profile: Main 10 (0x2)
[   0.296][v][vd] Requesting pixfmt 'videotoolbox_vld' from decoder.
[   0.498][i][vd] Using hardware decoding (videotoolbox).
[   0.498][v][vd] Decoder format: 3840x1608 videotoolbox[nv12] bt.2020-ncl/bt.2020/pq/limited/auto SP=13.600000 CL=unknown (auto 0.000000/0.000000/0.000000)
[   0.498][v][vd] Using container aspect ratio.
[   0.498][v][vf] [in] 3840x1608 videotoolbox[nv12] bt.2020-ncl/bt.2020/pq/limited/display SP=13.600000 CL=mpeg2/4/h264
[   0.498][v][vf] [userdeint] 3840x1608 videotoolbox[nv12] bt.2020-ncl/bt.2020/pq/limited/display SP=13.600000 CL=mpeg2/4/h264
[   0.498][v][vf] [userdeint] (disabled)
[   0.498][v][vf] [autorotate] 3840x1608 videotoolbox[nv12] bt.2020-ncl/bt.2020/pq/limited/display SP=13.600000 CL=mpeg2/4/h264
[   0.498][v][vf] [autorotate] (disabled)
[   0.498][v][vf] [convert] 3840x1608 videotoolbox[nv12] bt.2020-ncl/bt.2020/pq/limited/display SP=13.600000 CL=mpeg2/4/h264
[   0.498][v][vf] [convert] (disabled)
[   0.498][v][vf] [out] 3840x1608 videotoolbox[nv12] bt.2020-ncl/bt.2020/pq/limited/display SP=13.600000 CL=mpeg2/4/h264
[   0.498][v][af] [in] 48000Hz 7.1 8ch s32
[   0.498][v][af] [userspeed] 48000Hz 7.1 8ch s32
[   0.498][v][af] [userspeed] (disabled)
[   0.498][v][af] [convert] 48000Hz 7.1 8ch s32
```

- IINA log:
```
02:10:53.885 [iina][w] mpv log: [cplayer] warn: Audio/Video desynchronisation detected! Possible reasons include too slow
02:10:53.886 [iina][w] mpv log: [cplayer] warn: hardware, temporary CPU spikes, broken drivers, and broken files. Audio
02:10:53.886 [iina][w] mpv log: [cplayer] warn: position will not match to the video (see A-V status field).
02:10:53.886 [iina][w] mpv log: [cplayer] warn: 
```

**Steps to reproduce:**
Open a 10bit HEVC file 

- [ ] MPV does not have this problem

**How often does this happen?**
Always

**Comment by LeoNatan** [comment]
OK, so the problem seems more nuanced than originally thought. It seems when starting playback, acceleration works, but when seeking, then suddenly CPU usage goes to 300%. @inflation 
So not sure this is an upstream issue.

**Comment by inflation** [comment]
Do you enable exact seek in the Preference? In such mode, it would need to decode every frame from your current spot to your seek spot. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by LeoNatan** [comment]
I am familiar with that option, and I have not enabled it. it's the default "Keyframe seek". The problem doesn't occur on seek itself. Seek is instantaneous; the problem starts when playback resumes.

**Comment by LeoNatan** [comment]
Right, tested mpv better, and it's not an mpv issue. The problem is with IINA. In my mpv issue, I didn't run with `--hwdec=auto`, which didn't enable hardware acceleration. Also, seeking in mpv is not broken, like it is in IINA.

**Comment by LeoNatan** [comment]
Updated original issue post to state that mpv does not have this issue.

**Comment by LeoNatan** [comment]
Something else I noticed when seeking in IINA is that image "breaks" for a while until the next keyframe. This tells me that keyframe seek is not really working and IINA seeks wrong.

**Comment by LeoNatan** [comment]
Here is a screenshot of the seek issue. I think the loss of hardware acceleration happens after such a broken frame.

<img width="1094" alt="screen shot 2018-09-23 at 20 46 11" src="https://user-images.githubusercontent.com/2270433/45931128-de80e900-bf71-11e8-8f7c-0a39ccae3e40.png">

**Comment by inflation** [comment]
Then we need more log at the moment when it occurred. Unfortunately, I only have a 2016 MacBook Pro which only has hybrid hardware decoding on 10-Bit HEVC so I can't test it myself.

**Comment by LeoNatan** [comment]
I'll provide the logging information tomorrow. Thank you.

**Comment by LeoNatan** [comment]
Sorry for the constant flood of updates. It's an mpv issue. I finally managed to reproduce it with mpv. It happens considerably less, but exactly the same. I will open a new issue with mpv. Sorry.

**Comment by LeoNatan** [comment]
https://github.com/mpv-player/mpv/issues/6152
